test/tools/libtesteth/Options.cpp

```diff
@@ -99,6 +100,11 @@ Options::Options(int argc, char** argv)
 			if (i + 1 >= argc)
 				BOOST_THROW_EXCEPTION(InvalidOption(arg + " option is missing an argument."));
 		};
+		auto throwIfNotRandomTest = [&seenCreateRandomTest, &arg]()
+		{
+			if (!seenCreateRandomTest)
+				BOOST_THROW_EXCEPTION(InvalidOption(arg + " option should follow after --createRandomTest."));
+		};
```

Comment:
I believe `--debug` option can appear before `--createRandomTest`.  So the check should not happen during the parsing.  After the parsing, you can detect it.